src/sidebar/app/components/Footer.js

```diff
@@ -0,0 +1,333 @@
+import React from 'react';
+
+import {getPadStats} from '../editor.js';
+
+/**
+ * Formats time for the Notes footer
+ * @param time
+ * @returns {string}
+ */
+function formatFooterTime(date) {
+  date = date || Date.now();
+  return new Date(date).toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'});
+}
+
+/**
+ * Set animation on footerButtons toolbar
+ *
+ *
+ * @param {Boolean} animateSyncIcon Start looping animation on sync icon
+ * @param {Boolean} syncingLayout   if true, animate to syncingLayout (sync icon on right)
+ *                                  if false, animate to savingLayout (sync icon on left)
+ * @param {Boolean} warning         Apply yellow warning styling on toolbar
+ */
+function setAnimation( animateSyncIcon = true, syncingLayout, warning ) { // animateSyncIcon, syncingLayout, warning
+  const footerButtons = document.getElementById('footer-buttons');
+  const enableSync = document.getElementById('enable-sync');
+  const savingIndicator = document.getElementById('saving-indicator');
+
+  if (animateSyncIcon === true && !footerButtons.classList.contains('animateSyncIcon')) {
+    footerButtons.classList.add('animateSyncIcon');
+  } else if (animateSyncIcon === false && footerButtons.classList.contains('animateSyncIcon')) {
+    footerButtons.classList.remove('animateSyncIcon');
+  }
+
+  if (syncingLayout === true && footerButtons.classList.contains('savingLayout')) {
+    footerButtons.classList.replace('savingLayout', 'syncingLayout');
+    enableSync.style.backgroundColor = 'transparent';
+    // Start blink animation on saving-indicator
+    savingIndicator.classList.add('blink');
```

Comment:
are we gonna use any react magic for this animation or leave it like it is right now?

Reply:
First step was migration, refactoring is coming 👍